### PR TITLE
Add test dapp workflow template

### PR DESCRIPTION
**Pull request by katelynsills**
I'm hoping to test against a number of dapps in the agoric-sdk CI. To do so, I'd like to add a workflow template for testing dapps. Currently the only way to use it is to copy and paste and add the dapp name in [[INSERT_DAPP_NAME]] slots. Let me know if you know of a better way to do it.

Also changes the current workflow for testing dapp encouragement  and the documentation to fit the template.

### CODE REVIEW

**Comment by katelynsills** [comment]
Huh, there's a nonconformity. Some of our dapps are using `master`, others are using `main`. We should switch over to `main` for all, I think.

File: .github/workflow-templates/test-dapp.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # Select a branch on dapp to test against by adding text to the body of the
+    # pull request. For example: #dapp-encouragement-branch: zoe-release-0.7.0
+    # The default is 'master'
+    - name: Get the appropriate dapp branch
+      id: get-branch
+      uses: actions/github-script@0.9.0
+      with:
+        result-encoding: string
+        script: |
+          let branch = 'master';
```
Comment (Chris-Hibbert) [inline .github/workflow-templates/test-dapp.yml]: why isn't this 'main', too?

File: .github/workflows/test-documentation.yml [hunk 2]
```diff
@@ -31,16 +32,16 @@ jobs:
...
           ${{ runner.os }}-yarn-
 
-    # Select a branch on documentation to test against by adding text to the body of the
-    # pull request. For example: #documentation-branch: zoe-release-0.7.0
-    # The default is 'master'
-    - name: Get the appropriate documentation branch
+    # Select a branch on dapp to test against by adding text to the body of the
+    # pull request. For example: #dapp-encouragement-branch: zoe-release-0.7.0
+    # The default is 'main'
+    - name: Get the appropriate dapp branch
```
Comment (Chris-Hibbert) [inline .github/workflows/test-documentation.yml:38]: The names in the body changed from `documentation` to 'dapp', but the file name is still 'test-documentation.yml`.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]

File: .github/workflow-templates/test-dapp.yml (hunk 1 above)
Comment (katelynsills) [reply .github/workflow-templates/test-dapp.yml]: Good eye. Fixed

File: .github/workflows/test-documentation.yml (hunk 2 above)
Comment (katelynsills) [reply .github/workflows/test-documentation.yml:38]: Yes, it's testing the documentation using the dapp template which uses the term dapp generically

File: .github/workflow-templates/test-dapp.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/yarn
+        key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    # Select a branch on dapp to test against by adding text to the body of the
+    # pull request. For example: #dapp-encouragement-branch: zoe-release-0.7.0
+    # The default is 'master'
```
Comment (Chris-Hibbert) [inline .github/workflow-templates/test-dapp.yml]: missed this one.

**Review by Chris-Hibbert — APPROVED** [review]

**Pull request merged**